test/helpers/kubectl.go

```diff
@@ -4798,7 +4798,7 @@ func (kub *Kubectl) ensureKubectlVersion() error {
 		rcVersion = fmt.Sprintf("v%s.0", GetCurrentK8SEnv())
 	}
 	res = kub.Exec(
-		fmt.Sprintf("curl --output %s https://storage.googleapis.com/kubernetes-release/release/%s/bin/linux/amd64/kubectl && chmod +x %s",
+		fmt.Sprintf("curl --output %s https://dl.k8s.io/release/%s/bin/linux/amd64/kubectl && chmod +x %s",
```

Comment:
following redirect flag is required here, it takes me sometime to figure it out in https://github.com/cilium/cilium/pull/36235